File: app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Toast;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

public class MainActivity extends AppCompatActivity {

    private MainViewModel mainViewModel; // ViewModel 선언 (초기화는 onCreate에서)
A
    private EditText editTextId;
    private EditText editTextPassword;
    private Button buttonAction; // 로그인 또는 감정분석하기 버튼
    private Button buttonSignup;
    private Button buttonFindId;
    private Button buttonFindPassword;
    private LinearLayout layoutAuthHelpers; // 회원가입, 아이디/비밀번호 찾기 버튼 그룹

    private ActivityResultLauncher<Intent> loginActivityResultLauncher;

    public static final String PREFS_NAME = "MyLoginPrefs";
    public static final String KEY_IS_LOGGED_IN = "isLoggedIn";

    // MainActivity의 기본 생성자는 일반적으로 명시적으로 만들 필요가 없습니다.
    // public MainActivity() {
    // }

/* <<<<<<<<<<<<<<  ✨ Windsurf Command ⭐ >>>>>>>>>>>>>>>> */
    /**
     * Activity의 기본 생성자
     * <p> 이 메서드는 {@link MainViewModel}을 초기화하고, UI 요소를 초기화합니다.
     * 그리고 {@link ActivityResultContracts.StartActivityForResult()}를 사용하여 {@link LoginActivity}로부터 결과를 받습니다.
     * </p>
     * <p> 이 메서드는 SharedPreferences에서 이전 로그인 상태를 불러옵니다.
     * 그리고 로그인 성공/실패 결과에 따라 UI를 업데이트합니다.
     * </p>
/* <<<<<<<<<<  2ed79157-eca4-45de-95ec-fe2e34c0046e  >>>>>>>>>>> */
    @Override00000000
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // ViewModel 초기화 (올바른 위치)
        mainViewModel = new ViewModelProvider(this).get(MainViewModel.class);

        // UI 요소 초기화
        editTextId = findViewById(R.id.textid);
        editTextPassword = findViewById(R.id.textpassword);
        buttonAction = findViewById(R.id.menuBtn);
        buttonSignup = findViewById(R.id.signupBtn);
        buttonFindId = findViewById(R.id.passwordfindid);
        buttonFindPassword = findViewById(R.id.passwordfindBtn);

        // authHelperLayout ID를 가진 LinearLayout이 XML에 있는지 확인하고 연결합니다.
        // XML에 <LinearLayout android:id="@+id/authHelperLayout" ...> 와 같이 정의되어 있어야 합니다.
        View layoutView = findViewById(R.id.authHelperLayout); // 실제 XML ID로 변경 필요할 수 있음
        if (layoutView instanceof LinearLayout) {
            layoutAuthHelpers = (LinearLayout) layoutView;
        }

        // LoginActivity로부터 결과를 받기 위한 설정
        loginActivityResultLauncher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                result -> {
                    if (result.getResultCode() == RESULT_OK) {
                        // LoginActivity에서 로그인 성공 결과를 받음
                        mainViewModel.performLogin(); // ViewModel에 로그인 성공 알림
                        saveLoginState(true);    // SharedPreferences에 로그인 상태 저장
                        Toast.makeText(this, "로그인 되었습니다.", Toast.LENGTH_SHORT).show();
                    }
                });

        // ViewModel의 isLoggedIn LiveData 관찰하여 UI 업데이트
        mainViewModel.isLoggedIn.observe(this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean isLoggedIn) {
                updateUI(isLoggedIn);
            }
        });

        // 앱 시작 시 SharedPreferences에서 이전 로그인 상태 불러오기
        boolean isLoggedInPreviously = getLoginState();
        if (isLoggedInPreviously) {
            mainViewModel.performLogin(); // ViewModel 상태 동기화 (UI는 observe를 통해 자동 업데이트됨)
        } else {
            updateUI(false); // 로그아웃 상태 UI로 초기화 (필요시 호출, observe가 초기값도 처리)
        }

        // 초기 버튼 리스너 설정 (로그아웃 상태 기준)
        setupInitialButtonListeners();

    } // ### onCreate() 메소드 닫는 괄호 ###

    private void setupInitialButtonListeners() {
        // 이 함수는 로그인 상태에 따라 리스너 내용이 바뀌므로,
        // update

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    private val viewModel: MainViewModel by viewModels()
    private lateinit var loginLayout: View
    private lateinit var mainContentLayout: View
    private lateinit var bottomNav: BottomNavigationView
    private lateinit var editTextId: com.google.android.material.textfield.TextInputEditText
    private lateinit var editTextPassword: com.google.android.material.textfield.TextInputEditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 뷰 초기화
        loginLayout = findViewById(R.id.loginLayout)
        mainContentLayout = findViewById(R.id.mainContentLayout)
        bottomNav = findViewById(R.id.bottom_navigation)
        editTextId = findViewById(R.id.editId)
        editTextPassword = findViewById(R.id.editPassword)

        // 로그인 상태 관찰
        viewModel.isLoggedIn.observe(this, Observer { isLoggedIn ->
            if (isLoggedIn) {
                loginSuccess()
            } else {
                loginLayout.visibility = View.VISIBLE
                mainContentLayout.visibility = View.GONE
                bottomNav.visibility = View.GONE
            }
        })

        // 로그인 버튼 클릭 이벤트
        findViewById<Button>(R.id.btnLogin).setOnClickListener {
            attemptLogin()
        }

        // 음식 추천 버튼 클릭 이벤트
        findViewById<View>(R.id.btnRecommendFood).setOnClickListener {
            // EmotionActivity로 이동
            val intent = Intent(this, EmotionActivity::class.java)
            startActivity(intent)
        }

        // 회원가입 버튼 클릭 이벤트
        findViewById<View>(R.id.textSignUp).setOnClickListener {
            val intent = Intent(this, SignUpActivity::class.java)
            startActivity(intent)
        }

        // 하단 네비게이션 설정
        setupBottomNavigation()
    }

    private fun attemptLogin() {
        val id = editTextId.text.toString().trim()
        val password = editTextPassword.text.toString().trim()

        if (id.isEmpty() || password.isEmpty()) {
            Toast.makeText(this, "아이디와 비밀번호를 모두 입력해주세요.", Toast.LENGTH_SHORT).show()
            return
        }

        // 여기에 실제 로그인 로직 구현 (임시로 성공 처리)
        viewModel.performLogin()
        
        // 입력 필드 초기화
        editTextId.text?.clear()
        editTextPassword.text?.clear()
    }

    private fun loginSuccess() {
        // 로그인 성공 시 UI 업데이트
        loginLayout.visibility = View.GONE
        mainContentLayout.visibility = View.VISIBLE
        bottomNav.visibility = View.VISIBLE
        
        // 로그인 성공 토스트 메시지
        Toast.makeText(this, "로그인 되었습니다.", Toast.LENGTH_SHORT).show()
    }

    private fun setupBottomNavigation() {
        bottomNav.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.navigation_home -> {
                    // 홈 화면으로 이동
                    mainContentLayout.visibility = View.VISIBLE
                    true
                }
                R.id.navigation_history -> {
                    // 기록 화면으로 이동
                    // TODO: 기록 화면 구현 후 추가
                    true
                }
                R.id.navigation_profile -> {
                    // 마이페이지로 이동
                    // TODO: 마이페이지 화면 구현 후 추가
                    true
                }
                else -> false
            }
        }
    }
}
